test/quantization/core/test_quantized_op.py

```diff
@@ -6343,6 +6351,725 @@ def test_conv_transpose_reorder_issue_onednn(self):
+                strides,
+                pads,
+                dilations,
+                groups,
+                Y_scale,
+                Y_zero_point,
+                fp32_output,
+                post_op.binary_attr,
+                post_op.alpha,
+                post_op.unary_attr,
+                post_op.scalars,
+                post_op.algorithm,
+            )
+        else:
+            Y_q_cpu_tensor = qconv(
+                X_q_cpu_tensor,
+                X_scale,
+                X_zero_point,
+                packed_weight,
+                weight_scale,
+                weight_zero_point,
+                bias_float,
+                strides,
+                pads,
+                dilations,
+                groups,
+                Y_scale,
+                Y_zero_point,
+                fp32_output,
+                post_op.unary_attr,
+                post_op.scalars,
+                post_op.algorithm,
+            )
+            if fp32_output:
+                self.assertTrue(Y_q_cpu_tensor.dtype == torch.float32)
+                Y_q_cpu_tensor = torch.quantize_per_tensor(
+                    Y_q_cpu_tensor, scale=Y_scale, zero_point=Y_zero_point, dtype=output_dtype
+                ).int_repr()
+
+        # Make sure the results match
+        # assert_array_almost_equal compares using the following formula:
+        #     abs(desired-actual) < 1.5 * 10**(-decimal)
+        # (https://docs.scipy.org/doc/numpy/reference/generated/numpy.testing.assert_almost_equal.html)
+        # We use decimal = 0 to ignore off-by-1 differences between
+        # reference and test. Off-by-1 differences arise due to the order of
+        # round and zero_point addition operation, i.e., if addition
+        # followed by round is used by reference and round followed by
+        # addition is used by test, the results may differ by 1.
+        # For example, the result of round(2.5) + 1 is 3 while
+        # round(2.5 + 1) is 4 assuming the rounding mode is
+        # round-to-nearest, ties-to-even.
+
+        np.testing.assert_array_almost_equal(
+            result_ref_q.int_repr().cpu().numpy(),
+            Y_q_cpu_tensor.cpu().numpy(),
+            decimal=0,
+            err_msg=f"""X: {X_q}, W: {W_q}, b: {bias_float}, strides: {strides},
+            pads: {pads}, o_pads: {o_pads}, dilations: {dilations},
+            groups: {groups}, y_s: {Y_scale}, y_zp: {Y_zero_point}, X2: {X2_q}""",
+        )
+
+        # Return the quantized data for later reuse
+        return X_q, W_q, bias_float
+
+    @skipIfNoONEDNN
+    def test_qconv1d_pt2e(self):
+        batch_size = 3
+        groups_list = [1, 3]
+        input_channels_per_group = 2
+        output_channels_per_group = 2
+        length = 4
+        kernel = 3
+        stride = 1
+        pad = 1
+        dilation = 1
+        X_scale = 1.5
+        X_zero_point = 2
+        W_scale = [1.5]
+        W_zero_point = [0]
+        Y_scale = 4.2
+        Y_zero_point = 0
```

Comment:
if these are hardcoded could you move these to `_test_qconv_impl_cpu_tensor` to reduce the number of arguments

Reply:
Thanks for the suggestions. I have refactor the test cases to reduce duplicated arguments for the testing. I am afraid we still need to keep some necessary arguments due the difference between conv1d/2d/3d and different post ops to be fused.